### PR TITLE
yup: update to 0.24 and add setLocale

### PR DESCRIPTION
**Pull request by renchap**
* Add `setLocale` definition
* Add `less`, `more`, `notEqual` on `number` (according to https://github.com/jquense/yup/blob/master/CHANGELOG.md, doc is not up to date)

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jquense/yup/blob/master/src/customLocale.js & https://github.com/jquense/yup/blob/master/CHANGELOG.md
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/yup/index.d.ts [hunk 1]
```diff
@@ -89,6 +89,9 @@ export interface NumberSchemaConstructor {
 export interface NumberSchema extends Schema {
     min(limit: number | Ref, message?: string): NumberSchema;
     max(limit: number | Ref, message?: string): NumberSchema;
+    less(limit: number | Ref, message?: string): NumberSchema;
```
Comment (vtserman) [inline types/yup/index.d.ts]: It looks like it should be `lessThan`. see [number.js](https://github.com/jquense/yup/blob/v0.24.0/src/number.js)

File: types/yup/index.d.ts [hunk 2]
```diff
@@ -89,6 +89,9 @@ export interface NumberSchemaConstructor {
...
+    more(limit: number | Ref, message?: string): NumberSchema;
```
Comment (vtserman) [inline types/yup/index.d.ts]: Again, it should be `moreThan `.  see [number.js](https://github.com/jquense/yup/blob/1bc7b56f61d21c8681613acd9b531b9edd420879/src/number.js#L73)

**Review by dhardtke — APPROVED** [review]

File: types/yup/index.d.ts [hunk 3]
```diff
@@ -89,6 +89,9 @@ export interface NumberSchemaConstructor {
...
+    notEqual(limit: number | Ref, message?: string): NumberSchema;
```
Comment (vtserman) [inline types/yup/index.d.ts]: Regarding this addition, I couldn't find `notEqual` inside the [yup/number.js](https://github.com/jquense/yup/blob/v0.24.0/src/number.js). Looks like this line should be deleted.

**Review by vtserman — CHANGES_REQUESTED** [review]

File: types/yup/index.d.ts (hunk 3 above)
Comment (dhardtke) [reply types/yup/index.d.ts]: Dammit. Overlooked that... -.-

Comment (renchap) [reply types/yup/index.d.ts]: You are correct, and this commit also renamed `less` and `more`…
I will update the PR, and submit one in Yup repo to fix the Changelog.

Comment (renchap) [reply types/yup/index.d.ts]: Done, with a doc PR on Yup: https://github.com/jquense/yup/pull/167

**Review by vtserman — APPROVED** [review]
Looks good to me

File: types/yup/lib/customLocale.d.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import {
+  MixedSchema,
```
Comment (weswigham) [inline types/yup/lib/customLocale.d.ts]: Why all the unused imports? `LocaleObject` is the only type used in the file, so the other imports shouldn't be needed here.

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/yup/lib/customLocale.d.ts (hunk 4 above)
Comment (renchap) [reply types/yup/lib/customLocale.d.ts]: I defined `LocaleObject` in this file at first, and forgot to remove them after moving it.
This is fixed, thanks for catching this, I should have been more careful.

**Review by weswigham — APPROVED** [review]

**Pull request merged**